go/oasis-test-runner/test-runner.go

```diff
@@ -13,64 +16,69 @@ func main() {
-	// Byzantine merge node.
-	_ = cmd.Register(e2e.ByzantineMergeHonest)
-	_ = cmd.Register(e2e.ByzantineMergeWrong)
-	_ = cmd.Register(e2e.ByzantineMergeStraggler)
-	// Storage sync test.
-	_ = cmd.Register(e2e.StorageSync)
-	// Sentry test.
-	_ = cmd.Register(e2e.Sentry)
-	_ = cmd.Register(e2e.SentryEncryption)
-	// Keymanager restart test.
-	_ = cmd.Register(e2e.KeymanagerRestart)
-	// Dump/restore test.
-	_ = cmd.Register(e2e.DumpRestore)
-	// Halt test.
-	_ = cmd.Register(e2e.HaltRestore)
-	// Multiple runtimes test.
-	_ = cmd.Register(e2e.MultipleRuntimes)
-	// Registry CLI test.
-	_ = cmd.Register(e2e.RegistryCLI)
-	// Stake CLI test.
-	_ = cmd.Register(e2e.StakeCLI)
-	// Node shutdown test.
-	_ = cmd.Register(e2e.NodeShutdown)
-	// Gas fees tests.
-	_ = cmd.Register(e2e.GasFeesStaking)
-	_ = cmd.Register(e2e.GasFeesStakingDumpRestore)
-	_ = cmd.Register(e2e.GasFeesRuntimes)
-	// Identity CLI test.
-	_ = cmd.Register(e2e.IdentityCLI)
-	// Runtime prune test.
-	_ = cmd.Register(e2e.RuntimePrune)
-	// Runtime dynamic registration test.
-	_ = cmd.Register(e2e.RuntimeDynamic)
-	// Transaction source test.
-	_ = cmd.Register(e2e.TxSourceMultiShort)
-	_ = cmd.RegisterNondefault(e2e.TxSourceMulti)
-	// Node upgrade tests.
-	_ = cmd.Register(e2e.NodeUpgrade)
-	_ = cmd.Register(e2e.NodeUpgradeCancel)
-	// Debonding entries from genesis test.
-	_ = cmd.Register(e2e.Debond)
 
-	// Register the remote signer test cases.
-	rootCmd.Flags().AddFlagSet(remotesigner.Flags)
-	_ = cmd.Register(remotesigner.Basic)
+	// Register the all test cases.
+	cmd.RegisterTestParams(e2e.E2eParamsDummy.Name(), e2e.E2eParamsDummy.Parameters())
+	cmd.RegisterTestParams(e2e.RuntimeParamsDummy.Name(), e2e.RuntimeParamsDummy.Parameters())
+	cmd.RegisterTestParams(remotesigner.RemoteSignerParamsDummy.Name(), remotesigner.RemoteSignerParamsDummy.Parameters())
+	for _, s := range []scenario.Scenario{
+		// Runtime test.
+		e2e.Runtime,
+		e2e.RuntimeEncryption,
+		// Byzantine executor node.
+		e2e.ByzantineExecutorHonest,
+		e2e.ByzantineExecutorWrong,
+		e2e.ByzantineExecutorStraggler,
+		// Byzantine merge node.
+		e2e.ByzantineMergeHonest,
+		e2e.ByzantineMergeWrong,
+		e2e.ByzantineMergeStraggler,
+		// Storage sync test.
+		e2e.StorageSync,
+		// Sentry test.
+		e2e.Sentry,
+		e2e.SentryEncryption,
+		// Keymanager restart test.
+		e2e.KeymanagerRestart,
+		// Dump/restore test.
+		e2e.DumpRestore,
+		// Halt test.
+		e2e.HaltRestore,
+		// Multiple runtimes test.
+		e2e.MultipleRuntimes,
+		// Registry CLI test.
+		e2e.RegistryCLI,
+		// Stake CLI test.
+		e2e.StakeCLI,
+		// Node shutdown test.
+		e2e.NodeShutdown,
+		// Gas fees tests.
+		e2e.GasFeesStaking,
+		e2e.GasFeesStakingDumpRestore,
+		e2e.GasFeesRuntimes,
+		// Identity CLI test.
+		e2e.IdentityCLI,
+		// Runtime prune test.
+		e2e.RuntimePrune,
+		// Runtime dynamic registration test.
+		e2e.RuntimeDynamic,
+		// Transaction source test.
+		e2e.TxSourceMultiShort,
+		e2e.TxSourceMulti,
```

Comment:
This should be a non-default scenario as otherwise it will time out (it runs for 6 hours).